src/TTImp/Elab/RunElab.idr

```diff
@@ -88,7 +88,12 @@ elabScript fc nest env script@(NDCon nfc nm t ar args) exp
                       elabScript fc nest env
                               !(sc defs (toClosure withAll env
                                               !(quote defs env act'))) exp
-                  x => failWith defs $ "non-function RHS of a Bind: " ++ show x
+                  head => do -- we've got an eta-contracted form here. Eta-expand!
```

Comment:
This comment needs clarifying: we're not eta-expanding anything here, we're forming `k act'` (where `act'`
is `act` with `Elab` stripped off by `elabScript` I hope?)